### DISCUSSION TITLE
Referencing Variables in Vercel

### DISCUSSION THREAD

**Discussion by olliebatch** [vercel/community#3621] [Help] [answered]
### Summary

I'm trying to understand if the referencing of variables is supported in Vercel.

From the next docs: 
https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables#referencing-other-variables

Which allows you to reference other variables that you've defined . I was trying to use the system environment variable VERCEL_URL

`ROOT_URL="https://$VERCEL_URL"`

Is this approach supported in Vercel? 

When i actually use this ( as im using in a redirect it just attempts to go to the url without any other environment variables inside.

### Steps to Reproduce

Reference one environment variable with another and console log it.

**Comment by amyegan** [comment, marked as answer]
Hey @olliebatch. Where are you trying to include referenced environment variables? When I reference them in a `.env` file at the root of my project, they work as expected and I don't see any errors.

**Reply by olliebatch** [reply]
Hi @amyegan  i'm trying to reference variables in the vercel environment variables 
They do work fine in the `.env` but that approach doesn't work when vercel builds the project ( with referenced variables made on a preview branch)

**Reply by amyegan** [reply]
Vercel does not have support in the Environment Variables UI to reference other variables. Referencing other variables is only supported in `.env*` files. 

I was able to use a `.env` file in my Next.js test app to include `ROOT_URL=https://$VERCEL_URL`, which was able to pull in the correct URL for my preview deployment.

**Reply by rosariogueli** [reply]
does that mean that i have to commit my .env file and push it to the repository? Isn't that bad practice...
